src/Domain/Common/ValueObject.cs

```diff
@@ -31,8 +31,8 @@ public override bool Equals(object obj)
             }
 
             var other = (ValueObject)obj;
-            var thisValues = GetAtomicValues().GetEnumerator();
-            var otherValues = other.GetAtomicValues().GetEnumerator();
+            using var thisValues = GetAtomicValues().GetEnumerator();
+            using var otherValues = other.GetAtomicValues().GetEnumerator();
```

Comment:
What is the difference between `var thisValues` vs `using var thisValues` in this context?

Reply:
it's under `using` code block, so the objects will get disposed at the end of the parent code block.